packages/block-library/src/gallery/deprecated.js

```diff
@@ -15,6 +15,139 @@ import { RichText } from '@wordpress/block-editor';
 import { defaultColumnsNumber } from './shared';
 
 const deprecated = [
+	{
+		attributes: {
+			images: {
+				type: 'array',
+				default: [],
+				source: 'query',
+				selector: '.blocks-gallery-item',
+				query: {
+					url: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'src',
+					},
+					fullUrl: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-full-url',
+					},
+					link: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-link',
+					},
+					alt: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'alt',
+						default: '',
+					},
+					id: {
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-id',
+					},
+					caption: {
+						type: 'string',
+						source: 'html',
+						selector: '.blocks-gallery-item__caption',
+					},
+				},
+			},
+			ids: {
+				type: 'array',
+				default: [],
+			},
+			columns: {
+				type: 'number',
+			},
+			caption: {
+				type: 'string',
+				source: 'html',
+				selector: '.blocks-gallery-caption',
+			},
+			imageCrop: {
+				type: 'boolean',
+				default: true,
+			},
+			linkTo: {
+				type: 'string',
+				default: 'none',
+			},
+		},
+		isEligible( { ids } ) {
+			return ids &&
+				ids.length > 0 &&
+				ids.reduce( ( state, id ) => {
+					return state || ( typeof id === 'string' );
+				}, false );
```

Comment:
This is best described as 

```js
return ids.some( ( id ) => typeof id === 'string' );
```

Reply:
Thanks for this suggestion, I implemented it in: https://github.com/WordPress/gutenberg/pull/19163/commits/b0211d7354968ddcb822910457151f56a3135e7e